src/test/java/com/specialwarriors/conal/UserServiceTest.java

```diff
@@ -0,0 +1,111 @@
+package com.specialwarriors.conal;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import com.specialwarriors.conal.common.auth.oauth.GithubOAuth2WebClient;
+import com.specialwarriors.conal.common.exception.GeneralException;
+import com.specialwarriors.conal.user.domain.User;
+import com.specialwarriors.conal.user.repository.UserRepository;
+import com.specialwarriors.conal.user.service.UserService;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.data.redis.core.ValueOperations;
+
+@ExtendWith(MockitoExtension.class)
+public class UserServiceTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private RedisTemplate<String, String> redisTemplate;
+
+    @Mock
+    private GithubOAuth2WebClient githubOAuth2WebClient;
+
+    @InjectMocks
+    private UserService userService;
+
+    private User mockUser;
+
+    @BeforeEach
+    void setUp() {
+        mockUser = new User(1, "홍길동", "fdsf");
+    }
+
+    @Test
+    @DisplayName("유저 아이디로 사용자를 조회할 수 있다")
+    void findUserByUserId() {
+
+        // given
+        given(userRepository.findById(1L)).willReturn(Optional.of(mockUser));
+
+        // when
+        User result = userService.getUserByUserId(1L);
+
+        // then
+        assertThat(result).isEqualTo(mockUser);
+        verify(userRepository).findById(1L);
+    }
+
+    @Test
+    @DisplayName("유저 아이디로 사용자 조회 시 존재하지 않으면 예외를 던진다")
+    void throwsExceptionWhenUserNotFoundById() {
+
+        // given
+        given(userRepository.findById(1L)).willReturn(Optional.empty());
+
+        // when, then
+        assertThatThrownBy(() -> {
+            userService.getUserByUserId(1L);
+        }).isInstanceOf(GeneralException.class);
```

Comment:
```suggestion
        // when, then
        assertThatThrownBy(() -> userService.getUserByUserId(userId))
            .isInstanceOf(GeneralException.class)
            .extracting("exception")
            .isEqualTo(UserException.USER_NOT_FOUND);
```
- 람다로 처리할 수 있는 경우, 람다를 적극 활용하는 것은 어떨까요?
- 예외를 ENUM으로 관리하는 특성을 고려해 발생한 예외 ENUM까지 확인하면, 테스트 코드의 의도가 더 명확하지 않을까요?

Reply:
`발생한 예외 ENUM까지 확인`을 하는게 좋다고 생각했었는데 알려주셔서 감사합니다!